### PR TITLE
Split up DB models: part 1

### PR DESCRIPTION
**Pull request by jilleJr**
- \[x] I've added a new note in the `CHANGELOG.md` file, according to docs:
  https://iver-wharf.github.io/#/development/changelogs/writing-changelogs

## Summary

- Added packages
  - `pkg/model/database`
  - `pkg/model/request`
  - `pkg/model/response`
- Changed usage of `{type}Field{FieldName}` to `database.{Type}Fields.{FieldName}`
- Updated `token.go` to use new models
- Updated migrations to use new models
- Added transaction to PUT token and made it associate with the provider by the ID from the request JSON.

## Motivation

Related to #72, but doesn't close it yet as there are a lot more files to be dealt with.

This is like a taste sample. Better to get feedback on this early, as changing the remaining files will result in a giant PR that does pretty much the same stuff everywhere.

The -Search model will remain until we drop the backward compatibility, but only be used in the endpoints that will be deprecated in #71.

## Preview

The Swagger documentation now specifies more relevant models. I played around with the [`validate` tag from Swaggo](https://github.com/swaggo/swag#available) as well to mark some fields as required (the small red asterisk):

Screenshot from 2021-09-16 17-34-30
Screenshot from 2021-09-16 17-35-03
Screenshot from 2021-09-16 17-54-05

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -49,6 +49,20 @@ This project tries to follow [SemVer 2.0.0](https://semver.org/).
   data is slightly different; it has additional properties, and does not have a
   `status` property. (#43)
 
+- Added packages for "Plain Old Go Objects", with finer-grained decoupling
+  between database, HTTP request, and HTTP response models.
+  The Swagger documentation is affected by this, and some unused fields has
```
Comment (Alexamakans) [inline CHANGELOG.md:54]:
```suggestion
  The Swagger documentation is affected by this, and some unused fields have
```

**Review by Alexamakans — APPROVED** [review]
Good stuff 👍🏻 
Extra nice with the validations too 👍🏻

**Review by iverestefans — COMMENTED** [review]
LGTM!

**Review by fredx30 — COMMENTED** [review]
I like this idea of adding the required asterixes to swaggo. 
The idea of splitting up db models should terrify just about anyone. I dont see any explanation as to why this is a good idea. Elaborate?

**Comment by jilleJr** [comment]
> @fredx30  I like this idea of adding the required asterixes to swaggo.
> The idea of splitting up db models should terrify just about anyone. I dont see any explanation as to why this is a good idea. Elaborate?

Mentioned in the RFC's motivation. One of the points are this, which is relevant:

> - We use the database models in our HTTP request and response specs. The problems with this are:
>
>    - Security weakness as it induces a higher risk of exposing too much data to the user.
>
>   - Swagger documentation has room for improvement. We can elide irrelevant specifications such as allowing a project ID to be provided when creating a new project.
>
>   - The abstraction layers get mixed. With different models we would decouple the HTTP layer from the database layer, allowing for easier customization in both layers, as well as easier to understand as the models doesn’t have to worry both about how to serialize into JSON as well as their database table relations and constraints.
>
> https://iver-wharf.github.io/rfcs/published/0016-wharf-api-endpoints-cleanup#motivation

**Comment by jilleJr** [comment]
@iverestefans you only left a comment and not an "approve" review :)

**Comment by jilleJr** [comment]
@Alexamakans re-requested your review because of 461f49e & 45d00dd which was pushed after your review

**Review by Alexamakans — APPROVED** [review]

**Review by iverestefans — APPROVED** [review]
LGTM!

**Pull request merged**